frontend/packages/dev-console/src/actions/modify-application.ts

```diff
@@ -24,8 +25,8 @@ export const ModifyApplication = (kind: K8sKind, obj: K8sResourceKind): KebabOpt
 
 export const EditApplication = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
   return {
-    label: 'Edit',
-    href: `/edit?name=${obj.metadata.name}&kind=${obj.kind}`,
+    label: `Edit ${truncateText(obj.metadata.name)}`,
+    href: `/edit?name=${obj.metadata.name}&kind=${obj.kind || model.kind}`,
```

Comment:
I was just use `model.kind` always. But should we be using `referenceForModel`? `kind` isn't unique.

It sounds like the menu shouldn't have shown up, though, since the particular resource in the bug wasn't creating through the dev console "Add" flow.

Reply:
It shouldn't have. I've removed it at https://github.com/openshift/console/pull/4207/files#diff-2c9504a8d75ebb51fe3d613cf73328f2L8

Reply:
I guess I'm not clear on the fix in that case. Should we be looking at an annotation or label on the resource to know it was created from the console add flow? I see the Edit action showing up for deployments and other resources that were created outside console.

Reply:
Possibly we should. This PR does not fix that only the know crashing issue. I can file a separate issue for that or fix as part of this PR if you'd prefer.

Reply:
OK, I'm good tracking it as a separate BZ.

Reply:
https://issues.redhat.com/browse/ODC-2975